site/DataPageV2.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+    getVariableDataRoute,
+    getVariableMetadataRoute,
+    GrapherInterface,
+    GrapherProgrammaticInterface,
+} from "@ourworldindata/grapher"
+import {
+    uniq,
+    SiteFooterContext,
+    MarkdownTextWrap,
+    DataPageDataV2,
+    serializeJSONForHTML,
+    mergePartialGrapherConfigs,
+    compact,
+    FaqEntryData,
+} from "@ourworldindata/utils"
+import React from "react"
+import urljoin from "url-join"
+import {
+    ADMIN_BASE_URL,
+    BAKED_GRAPHER_URL,
+    DATA_API_URL,
+} from "../settings/clientSettings.js"
+import {
+    DataPageV2Content,
+    OWID_DATAPAGE_CONTENT_ROOT_ID,
+} from "./DataPageV2Content.js"
+import { Head } from "./Head.js"
+import { SiteFooter } from "./SiteFooter.js"
+import { SiteHeader } from "./SiteHeader.js"
+import { IFrameDetector } from "./IframeDetector.js"
+import { DebugProvider } from "./gdocs/DebugContext.js"
+
+export const DataPageV2 = (props: {
+    grapher: GrapherInterface | undefined
+    datapageData: DataPageDataV2
+    baseUrl: string
+    baseGrapherUrl: string
+    isPreviewing: boolean
+    faqEntries?: FaqEntryData
+}) => {
+    const {
+        grapher,
+        datapageData,
+        baseGrapherUrl,
+        baseUrl,
+        isPreviewing,
+        faqEntries,
+    } = props
+    const pageTitle = grapher?.title ?? datapageData.title
+    const canonicalUrl = grapher?.slug
+        ? urljoin(baseGrapherUrl, grapher.slug as string)
+        : ""
+    let pageDesc: string
+    if (grapher?.subtitle?.length) {
+        // convert subtitle from markdown to plaintext
+        pageDesc = new MarkdownTextWrap({
+            text: grapher.subtitle,
+            fontSize: 10,
+        }).plaintext
+    } else pageDesc = "An interactive visualization from Our World in Data."
+
+    // Due to thumbnails not taking into account URL parameters, they are often inaccurate on
+    // social media. We decided to remove them and use a single thumbnail for all charts.
+    // See https://github.com/owid/owid-grapher/issues/1086
+    //
+    // const imageUrl = urljoin(
+    //     baseGrapherUrl,
+    //     "exports",
+    //     `${grapher.slug}.png?v=${grapher.version}`
+    // )
+    const imageUrl: string = urljoin(baseUrl, "default-grapher-thumbnail.png")
+    const imageWidth = "1200"
+    const imageHeight = "628"
+
+    const variableIds: number[] = uniq(
+        compact(grapher?.dimensions?.map((d) => d.variableId))
+    )
+
+    const mergedGrapherConfig = mergePartialGrapherConfigs(
+        datapageData.chartConfig as GrapherInterface,
```

Comment:
The types are different here because we have not lifted the type definitions up to the types project yet, so this cast is still required unfortunately